app/services/segment_delete_service.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class SegmentDeleteService
+  SEGMENT_URI = URI('https://gdpr.segment.com/graphql')
+
+  class SegmentDeleteServiceError < StandardError; end
+  class MissingAuthenticationData < SegmentDeleteServiceError; end
+  class SegmentAuthenticationError < SegmentDeleteServiceError; end
+  class SegmentUserDeletionError < SegmentDeleteServiceError; end
+
+  # TODO: password in ENV :) and raise error if it is not there
```

Comment:
I intend to do this right now, don't panic 😝

Reply:
Also raising an error is only if the feature is enabled (the config is provided but it is not correct)